internal/services/datadog/azurerm_datadog_monitor_sso_configuration.go

```diff
@@ -63,15 +64,18 @@ func resourceDatadogSingleSignOnConfigurations() *pluginsdk.Resource {
 				ValidateFunc: validate.DatadogEnterpriseApplicationID,
 			},
 
-			"single_sign_on_enabled": {
+			"single_sign_on": {
 				Type:     pluginsdk.TypeString,
-				Required: true,
+				Required: features.FivePointOhBeta(),
+				Optional: !features.FivePointOhBeta(),
+				Computed: !features.FivePointOhBeta(),
```

Comment:
These should be set correctly for the 5.0 code path and overridden in the feature flagged section below to ease the burden of removing them later. (which will also side-step the lint error flagged here)